### ISSUE TITLE
"object '.doSnowGlobals' not found" error

### ISSUE THREAD

**Issue by adamlilith** [adamlilith/enmSdm#141]
This is largely a resolved issue, but I'm posting it here because it occasionally sometimes crops up when using functions that can do multicoring, like `trainMaxEnt` and `bioticVelocity`. When you try using >1 core, sometimes on some machines you can get the error

`object '.doSnowGlobals' not found`

I have, for example, received this error on a machine that uses the exact same set of libraries as another, and yet it worked on the second. This StackExchange [post](https://stackoverflow.com/questions/25079527/error-object-dosnowglobals-not-found) explains the issue, or so it seems, but in my case none of the fixes fixed it, and if you're not interested in looking inside the function, it wouldn't help anyway.

What has worked for me:
* An aggressive virus checker can disable access to cores. This was probably my issue, but I don't have admin access to change the antivirus settings.
* However, uninstalling R, then reinstalling (even the same version) seemed to work.

Yes, it was that "simple", and it may not work for you, but it's a solution which I did not find documented on numerous help boards.

Best,
Adam

**Comment by adamlilith** [comment]
And as I am finding, sometimes this does not work. The issue occurs when I issue `setwd(stuff)`, I get the `object '.doSnowGlobals' not found` error. The default directory also contains my R "libraries" folder, so maybe that's the issue?

Fixes found so far (that do not work for me):
* Registering `doParallel` with each node: StackExchange [post](https://stackoverflow.com/questions/25079527/error-object-dosnowglobals-not-found)
* Issue with having a custom R `libraries` folder not getting passed to each node: GH [issue](https://github.com/SimoneTiberi/distinct/issues/4)

**Comment by adamlilith** [comment]
Fixed!!!

The problem was that my custom library path wasn't being passed to each node. This was explained in this SE [post](https://stackoverflow.com/questions/25079527/error-object-dosnowglobals-not-found), but the exact fix wasn't actually spelled out. In my case, I needed to get the .libPaths() strings before sending jobs to nodes, pass these strings to the nodes with the jobs, then in the worker function set .libPaths(myStrings).

This is all under the hood, but consumed enough time to document what didn't work.
